PANA - funnel docs edit (#31994)
DataDog/documentation: content/en/product_analytics/charts/funnel_analysis.md

{{< img src="product_analytics/journeys/funnel_analysis/pana_funnel_video1.mp4" alt="Filtering network map with search" video=true >}}

> If you have a starting point in mind, but aren't sure what your users did next, the funnel step editor automatically loads the top most common **views** and **actions** that users typically see.

Reviewer: ```suggestion
If you aren't sure what your users did, the funnel step editor automatically loads common **views** and **actions** that you can choose to add as steps.
```
I just noticed that you can _start_ a funnel with a view or action, which made the "what your users did next" thing make a little less sense!

{{< img src="product_analytics/journeys/funnel_analysis/pana_funnel_dropoffs.png" alt="The funnel step editor automatically loads the top most common views and actions that users typically see and take next." style="width:50%;" >}}

### Add filters

> When constructing your funnel, you can filter globally or on a step.

Reviewer: ```suggestion
When constructing your funnel, you can filter users globally, or for a specific step.
```

{{< img src="product_analytics/journeys/funnel_analysis/pana_funnel_video1.mp4" alt="Filtering network map with search" video=true >}}

> If you have a starting point in mind, but aren't sure what your users did next, the funnel step editor automatically loads the top most common **views** and **actions** that users typically see.

Reviewer 2: I think we don't have this anymore

{{< img src="product_analytics/journeys/funnel_analysis/pana_funnel_dropoffs.png" alt="The funnel step editor automatically loads the top most common views and actions that users typically see and take next." style="width:50%;" >}}

> <div class="alert alert-info"> Any action or view that happens between two steps in a funnel does not impact the step-by-step or overall conversion rate. As long as step 1 and step 2 happen in the right order in a given session at least once, it counts as a single converted session.</div>

Reviewer: ```suggestion
<div class="alert alert-info"> Any action or view between two steps in a funnel does not impact the step-by-step or overall conversion rate. As long as step 1 and step 2 happen in the right order in a given session at least once, it counts as a single converted session.</div>
```

Reviewer: Now that I think about it, maybe this would make more sense in the section about conversion computing?

Click on the funnel chart to open a side panel with additional context about user dropoffs. Then, navigate between steps to see:

- **Step performance**: See metrics related to conversion (for example, conversion rate, dropped off sessions, and average conversion time).
- **Branching paths from the previous step**: Discover the other paths that users take instead of the ones you anticipate. This section is only available when looking at a step between two views in a funnel.
- **Issues that might be affecting conversion**: See the top issues that might be affecting conversions and investigate them further with the links to RUM, Error Tracking, and Session Replay.

{{< img src="product_analytics/journeys/funnel_analysis/pana_funnel_metrics.png" alt="Click on the funnel view to see additional context about user dropoffs." style="width:90%;" >}}

> ### Filtering

Reviewer: I'd like to talk a bit about headings and parts of speech! This is a subtle thing that is hard to get consistent throughout an entire docs set, but once you know to look for it, it does help you think about how you're directing users' attention in your docs.

For headings, I like to format them as **verb phrases** in the **imperative mood** wherever I can. Wow, sounds nerdy! Let's break that down:
- **Verb phrases**: Verbs are useful because we're telling users about something they can _do_. It might be correct to call this section "Filters," but that might leave a user wondering, well, what about filters? Creating them? Deleting them? Adding those verbs there helps you understand exactly what's going on. And verb phrases are just adding context to the verb: "create" would be bizarre on its own, but "create a filter"? BABY YOU GOT A STEW GOING.
- **Imperative mood**: This just refers to how you would phrase it if you were directing someone to do something. People use gerunds, as in "creating a filter," as headings in documentation sometimes, but "create a filter" - as if you're pointing at someone on the street and being like "hey it's an emergency and you need to do something! Create a filter right now!!" - cuts to the chase a little quicker and makes things read, I dunno, a little more purposefully to me? 

The choices you make here are largely dependent on personal preferences and style, but regardless, it's a good idea to stay as internally consistent as you can 🙂

You can analyze conversion by **session**, **user** or **account**. This can be useful if you suspect, for instance, that a minority of your user base converts at a high rate.

- If you select <strong>Session</strong>, all steps must be completed within the same <code>@session.id</code> to count as a conversion.

- If you select <strong>User</strong>, the funnel requires the same individual user (<code>@user.id</code>) to complete every step for the conversion to count.

- If you select <strong>Account</strong>, different users within the same account can complete different steps and the conversion still counts. In this case, the funnel is tied to the <code>@account.id</code> facet.

{{< img src="product_analytics/journeys/funnel_analysis/pana_funnel_refine_conversion.png" alt="The section of the UI where you can select Session, User, or Account to analyze your conversion.." style="width:50%;" >}}
 

You can measure conversion by the following attributes:

- **Conversion count**: A count of users who went through the funnel you've defined.

> - **Conversion rate**: A conversion refers to the moment when a user responds to a call to action. This rate is the percentage of users who have entered the funnel and converted.

Reviewer: ```suggestion
- **Conversion rate**: The percentage of users who entered the funnel and converted.
```

{{< img src="product_analytics/journeys/funnel_analysis/pana_funnel_conversion.png" alt="Measure attributes across all steps or specific steps." style="width:60%;" >}}

Use the [**filter**](#filtering) selector to scope to the various criteria you want to define. These filters are applied to all steps in your funnel.

Next, click a datapoint to **investigate the specific attributes** that might have affected conversion rates, such as page load speed, ease of navigation, or checkout experience.

## Conversion computing metrics

### How Datadog computes conversion metrics
Consider a funnel with events `A → B → C` and event steps `A, A, A, B, C, C`. 

In this case, Datadog counts one conversion. This is because the conversion calculation matches only the first occurrence of event **A** and the first occurrence of event **C** in the sequence. 

To further illustrate, if the user performs the event sequence `A, A, A, B, C, C, A, B, C`, Datadog counts two conversions. The first conversion completes with the sequence `A, A, A, B, C, C`, and the second conversion begins with the next occurrence of `A`.

> Datadog calculates the average time between steps based on the average duration between the first and last step of each conversion.

Reviewer: ```suggestion
Datadog calculates the average time between steps by averaging the total duration between the first and last step of each conversion over the total number of steps.
```

> If you analyze your funnel by **user** or by **Account**, you can define your conversion time frame in hours or days of the first event. The default time frame for conversions is one day (a 24-hour window, not a calendar date) to determine if a conversion happened or not.

Reviewer: ```suggestion
If you analyze your funnel by **user** or by **account**, you can define your conversion time frame in hours or days since the first event. The default time frame for conversions is one day (a 24-hour window, not a calendar date) to determine if a conversion happened.
```

> <div class="alert alert-info"> Any action or view that happens between two steps in a funnel does not impact the step-by-step or overall conversion rate. As long as step 1 and step 2 happen in the right order in a given session at least once, it counts as a single converted session.</div>

Reviewer: ```suggestion
<div class="alert alert-info"> Any action or view that happens between two steps in a funnel does not impact the step-by-step or overall conversion rate. As long as steps A and C happen in the right order in a given session at least once, it counts as a single converted session.</div>
```

> <div class="alert alert-info"> Any action or view that happens between two steps in a funnel does not impact the step-by-step or overall conversion rate. As long as step 1 and step 2 happen in the right order in a given session at least once, it counts as a single converted session.</div>

Reviewer: I would put it a couple paragraphs up! That way, you can group it with the other paragraphs where you're talking about step orders, rather than the ones where you're talking about time.

### Choose a conversion counting method

> When computing your conversion, you can choose between two counting methods: **Unique** or **Total**. 

Reviewer: Notice that here, you've capitalized your options, but you haven't earlier in the page (for example, "session, user, or account"). I would pick one way of doing it and stay consistent! It doesn't really matter which you choose, but as you self-edit, this is a great type of inconsistency to keep an eye out for.

- **Unique**: Counts conversion only once per session, user, or account. For example, if the user completes the funnel sequence `A → B → C` multiple times within the same session (for example, `A, B, C, A, B, C`), it counts as **one conversion**. The `Unique` setting counts only the first conversion per session (or per user, depending on your analysis scope).
<br> 
<br> 

- **Total**: Counts a conversion each time the same session ID, user or account completes the defined funnel. Using the same example (`A, B, C, A, B, C`), this methods counts **two conversions**. The `Total` setting doesn't multiply conversions for repeated steps within a single flow, instead it counts complete flows, not the number of times an intermediate step is repeated. 

To choose how conversions are counted in your funnel, beside **Conversion count**, select **Unique** or **Total**,

{{< img src="product_analytics/journeys/funnel_analysis/funnel_analysis_conversion.png" alt="Select a conversion measure, whether Unique or Total, to determine how your session conversions are counted." style="width:80%;" >}}

> ## Change the visualization

Reviewer: You talk about visualization types earlier in the page, so it would be a good idea to link down to here in case a user isn't familiar with what they are!

Use the Group by dropdown to group the data by a specific attribute.

> **Note**: `Group by` does not work with the funnel steps visualization, your visualization automatically changes to a top list.

Reviewer: Ah, thanks! Then I would say something like:
```suggestion
**Note**: `Group by` does not work with the funnel steps visualization. If you select it, your visualization automatically changes to a top list.
```

## Refine conversion

You can further analyze the information on the funnel page to understand the conversion rate. Conversion rate is a crucial metric that measures the effectiveness of your site or application.

You can analyze conversion by **session**, **users** or **accounts**, which means you can understand conversion across all sessions, or by distinct users or accounts. This can be useful if you suspect, for instance, that a minority of your user base converts at a high rate.

{{< img src="product_analytics/journeys/funnel_analysis/pana_funnel_refine_conversion.png" alt="Specify which facet you'd like to use to refine your conversion. The available options are Session, User, and Account." style="width:50%;" >}}

<div class="alert alert-info"> 

> If you select <code>Session</code>, all steps must be completed within the same <code>@session.id</code> to count as a conversion.

Reviewer: I would use bold to refer to dropdown selections throughout this topic rather than monospace. That way, it's harder for customers to get it confused with something they configured somewhere in their code.

If you select <code>User</code>, the funnel requires the same individual user (<code>@user.id</code>) to complete every step for the conversion to count.

> If you select <code>Account</code>, different users within the same account can complete different steps and the conversion still counts. In this case, the funnel is tied to the <code>@account.id</code> facet which maps to org IDs in Datadog.  

Reviewer: ```suggestion
If you select <code>Account</code>, different users within the same account can complete different steps and the conversion still counts. In this case, the funnel is tied to the <code>@account.id</code> facet, which maps to org IDs in Datadog.  
```

</div>

You can measure conversion by the following attributes:

- **Conversion count**: A count of users who went through the funnel you've defined.
- **Conversion rate**: A conversion refers to the moment when a user responds to a call to action. This rate is the percentage of users who have entered the funnel and converted.

>   **Note**: Only completed RUM sessions contribute to conversion rates.

Reviewer 3: I don't think this note applies anymore

- **Time to convert**: The time it took for the user to complete the step events.

You can measure these attributes **across all steps** or between **specific steps**.

{{< img src="product_analytics/journeys/funnel_analysis/pana_funnel_conversion.png" alt="Measure attributes across all steps or specific steps." style="width:60%;" >}}

> Use the **filter** selector to filter by various criteria that you define. These filters are applied to all steps of the funnel.

Reviewer: ```suggestion
Use the **filter** selector to filter by various criteria that you define. These filters are applied to all steps in your funnel.
```
Very minor, just noticed that the language changed a very little bit!

Next, click a datapoint to **investigate the specific attributes** that might have affected conversion rates, such as page load speed, ease of navigation, or checkout experience.

## How Datadog computes conversion metrics

Consider a funnel with events `A → B → C` and event steps: 

```
A, A, A, B, C, C
```

If a user performs the actions as in the example above, Datadog counts it as one conversion. This is because the conversion calculations only look at the first element **A** matched and the first element **C** matched. 

<div class="alert alert-info">
If the user performs the event steps <code>A, A, A, B, C, C, A,...</code>, it would count as two conversions. The first conversion ends at <code>A, A, A, B, C, C</code>, and the second conversion begins with <code>A,...</code>

</div>

> {{< img src="product_analytics/journeys/funnel_analysis/funnel_analysis_conversion.png" alt="Select a conversion measure, whether Unique or Total, to determine how your session conversions are counted." style="width:80%;" >}}

Reviewer: This screenshot doesn't have any context associated with it, so I wasn't initially sure what to do with this information. There was a pretty sharp transition between conceptual and procedural content - maybe add in some steps to smooth that transition so users know when they're supposed to be learning vs. when they're supposed to be doing something in the UI?

> **Unique**: This means that your conversion (session, user, or account) is counted only once per session, user or account. So, if the user performs the actions A → B → C → A → B → C during the session or time frame, it counts as 1 conversion.

Reviewer: ```suggestion
**Unique**: This means that your conversion (session, user, or account) is counted only once per session, user or account. So, if the user performs the actions A → B → C → A → B → C during the session or time frame, it counts as one conversion.
```
Also, we spell out single-digit numbers 🙂

**Total**: This means that your conversion (session, user, or account) is each time the user completes the defined funnel. For example, if the user performs the actions A → B → C → A → B → C during the session or time frame, it counts as 2 conversions.

> If you analyze your funnel by `user` or by `account`, you can define your conversion timeframe in hours or days of the first event. The default timeframe for conversions is 1 day (a 24-hour window not calendar dates) to determine if a conversion happened or not.

Reviewer: ```suggestion
If you analyze your funnel by `user` or by `account`, you can define your conversion time frame in hours or days of the first event. The default time frame for conversions is one day (a 24-hour window, not a calendar date).
```

> The average time between steps is calculated by looking at the average duration between each conversion and takes all conversions into account, no matter if `unique` or `total` is selected for the counts.

Reviewer: ```suggestion
The average time between steps is calculated by looking at the average duration between each conversion and takes all conversions into account, regardless of whether you selected `unique` or `total` for the counts.
```

> The **conversion rate** compares the total number of conversions to the total number of users, accounts, or sessions that enter the funnel.

Reviewer: ```suggestion
```
Looks like we explain this in the previous section, so I think you're off the hook explaining it again here!

Use the Group by dropdown to group the data by a specific attribute.

**Note**: The `Group by` option is not supported for the funnel steps visualization; when applied, the visualization automatically changes to a top list.

## Refine conversion

You can further analyze the information on the funnel page to understand the conversion rate. Conversion rate is a crucial metric that measures the effectiveness of your site or application.

You can analyze conversion by **session**, **users** or **accounts**. This can be useful if you suspect, for instance, that a minority of your user base converts at a high rate.

- If you select <strong>Session</strong>, all steps must be completed within the same <code>@session.id</code> to count as a conversion.

- If you select <strong>User</strong>, the funnel requires the same individual user (<code>@user.id</code>) to complete every step for the conversion to count.

- If you select <strong>Account</strong>, different users within the same account can complete different steps and the conversion still counts. In this case, the funnel is tied to the <code>@account.id</code> facet.

> {{< img src="product_analytics/journeys/funnel_analysis/pana_funnel_refine_conversion.png" alt="Specify which facet you'd like to use to refine your conversion. The available options are Session, User, and Account." style="width:50%;" >}}

Reviewer: ```suggestion
{{< img src="product_analytics/journeys/funnel_analysis/pana_funnel_refine_conversion.png" alt="The section of the UI where you can select Session, User, or Account to analyze your conversion." style="width:50%;" >}}
```
I would make alt text a description of what the screenshot is, rather than providing an instruction. Alt text exists to provide an equivalent experience for users who can't see the image (due to things like visual impairments or internet connectivity issues), so all you have to do is close that gap and you're set! A good way of thinking about this is imagining you're telling a friend on the phone about it 🙂

 

You can measure conversion by the following attributes:

- **Conversion count**: A count of users who went through the funnel you've defined.
- **Conversion rate**: A conversion refers to the moment when a user responds to a call to action. This rate is the percentage of users who have entered the funnel and converted.

> - **Time to convert**: The time it took for the user to complete the step events. This is not available for the _Steps_ visualization, it changes to timeseries when selected.

Reviewer: ```suggestion
- **Time to convert**: The time it took for the user to complete the step events. This is not available for the steps visualization, it changes to timeseries when selected.
```

It looks like this is the only time a visualization name is italicized and capitalized on this page. It might be worth looking to see if it's something we do elsewhere in the Product Analytics docs, if it doesn't happen elsewhere on this page.

Text formatting is one of those things where sometimes we have very well-defined rules, and other times we don't. When you're handling a situation where we don't, I'd think about what you're trying to tell your user with the formatting (for example, when we bold the name of a button that you're asking the user to click, we do so to make the label easier for them to spot when they're skimming). And sometimes we'll capitalize things in the UI to indicate that they're Very Important!

If you don't really have a reason to make something Look Important, like talking about a thing that a user has to do in the UI, I'd just leave it in sentence case. It's still understandable, but for one thing, you're doing less work, and for another, you're not making the user do quite as much work to pick out the important stuff.

{{< img src="product_analytics/journeys/funnel_analysis/pana_funnel_refine_conversion.png" alt="Specify which facet you'd like to use to refine your conversion. The available options are Session, User, and Account." style="width:50%;" >}}
 

You can measure conversion by the following attributes:

- **Conversion count**: A count of users who went through the funnel you've defined.
- **Conversion rate**: A conversion refers to the moment when a user responds to a call to action. This rate is the percentage of users who have entered the funnel and converted.
- **Time to convert**: The time it took for the user to complete the step events. This is not available for the _Steps_ visualization, it changes to timeseries when selected.

You can measure these attributes **across all steps** or between **specific steps**.

{{< img src="product_analytics/journeys/funnel_analysis/pana_funnel_conversion.png" alt="Measure attributes across all steps or specific steps." style="width:60%;" >}}

> Use the **filter** selector to scope to the various criteria you want to define. These filters are applied to all steps in your funnel.

Reviewer: Does this refer to a different UI from the "Filtering" section? If so, I would explain it further. If not, I would either take this out, or link to that section. What we want to avoid here is confusing your user by making them wonder if they've been able to follow the way you've described the UI - we don't want them to have to mentally jump around to figure that out 🙂

Next, click a datapoint to **investigate the specific attributes** that might have affected conversion rates, such as page load speed, ease of navigation, or checkout experience.

## Conversion computing metrics

### How Datadog computes conversion metrics
Consider a funnel with events `A → B → C` and event steps: 

```
A, A, A, B, C, C
```

> In this case, Datadog counts one conversion. This is because the conversion calculations matches only the first occurence of event **A** and the first occurence of event **C** in the sequence. 

Reviewer: ```suggestion
In this case, Datadog counts one conversion. This is because the conversion calculation matches only the first occurrence of event **A** and the first occurrence of event **C** in the sequence. 
```

<div class="alert alert-info">
To further illustrate, if the user performs the event sequence <code>A, A, A, B, C, C, A, B, C...</code>, Datadog counts two conversions. The first conversion completes with the sequence <code>A, A, A, B, C, C</code>, and the second conversion begins with the next occurence of <code>A</code>.
</div>

> The average time between steps is calculated based on the average duration between the first and last step of each conversion. This calculation takes all conversions into account regardless of whether you selected `unique` or `total` for the counts.

Reviewer: ```suggestion
Datadog calculates the average time between steps based on the average duration between the first and last step of each conversion.
```
I would move the thing about selecting Unique or Total for the counts to the next section. You haven't explained it yet, so your user doesn't know what you're talking about yet! And they could get confused because the past tense implied that they'd already done it.

(Also, to bring this back to what I was saying earlier about formatting - notice how in the next section it's formatted in bold, rather than monospace, which in my mind would be more appropriate for something a user has to type into a terminal, for example.)

### Choosing a conversion counting method

When computing your conversion, you can choose between two counting methods: **Unique** or **Total**. 

- **Unique**: Counts conversion only once per session, user, or account. For example, if the user completes the funnel sequence `A → B → C` multiple times within the same session (for example, `A → B → C → A → B → C`), it counts as **one conversion**.

> - **Total**: Counts a conversion each time the same session, user or account completes the defined funnel. Using the same example (`A → B → C → A → B → C`), this methods counts **two conversions**.

Reviewer: ```suggestion
- **Total**: Counts a conversion each time the same session, user, or account completes the defined funnel. Using the same example (`A → B → C → A → B → C`), this methods counts **two conversions**.
```

Reviewer: This list is a little odd if you read it to mean "each time the same session completes the defined funnel" - I wouldn't think of a session as something capable of completing a funnel.

Author: Got it! A successful pass through the funnel, which denotes a conversion, can be attributed to session IDs as well. In the case, then, would session ID makes more sense?

{{< img src="product_analytics/journeys/funnel_analysis/funnel_analysis_conversion.png" alt="Select a conversion measure, whether Unique or Total, to determine how your session conversions are counted." style="width:80%;" >}}

<div class="alert alert-info">
The <strong>Total</strong> setting doesn't multiply conversions for repeated steps within a single flow, instead it counts complete flows, not the number of times an intermediate step is repeated. 

<br>
<br

If you switch to <strong>Unique</strong>, then only the first conversion per session (or per user, depending on your analysis scope) is counted.

> </div>

Reviewer: This information should be incorporated into where you explained what both of these things mean above. Otherwise, a user might miss this and miss key information. When you put an explanation all in one place, that's far less likely to happen!

(Also, a note about structure, even though it won't apply in the final product here - if you explain "unique," followed by "total," in one part of your documentation, you should refer to them in the same order at other times. This helps customers structure information in their heads! It's called parallelism.)

> If you analyze your funnel by `user` or by `account`, you can define your conversion time frame in hours or days of the first event. The default time frame for conversions is onw day (a 24-hour window, not a calendar date) to determine if a conversion happened or not.

Reviewer: ```suggestion
If you analyze your funnel by **User** or by **Account**, you can define your conversion time frame in hours or days of the first event. The default time frame for conversions is one day (a 24-hour window, not a calendar date) to determine if a conversion happened.
```
I'm not sure this is the right place to put this. It feels like a bit of a non-sequitur, because I don't know how it fits in with the information that precedes it.

Author: Moved this to the end of "How Datadog computes conversion metrics" section. Let me know what you think

- If you select <strong>Session</strong>, all steps must be completed within the same <code>@session.id</code> to count as a conversion.

- If you select <strong>User</strong>, the funnel requires the same individual user (<code>@user.id</code>) to complete every step for the conversion to count.

- If you select <strong>Account</strong>, different users within the same account can complete different steps and the conversion still counts. In this case, the funnel is tied to the <code>@account.id</code> facet, which maps to org IDs in Datadog. 

{{< img src="product_analytics/journeys/funnel_analysis/pana_funnel_refine_conversion.png" alt="Specify which facet you'd like to use to refine your conversion. The available options are Session, User, and Account." style="width:50%;" >}}
 

You can measure conversion by the following attributes:

- **Conversion count**: A count of users who went through the funnel you've defined.
- **Conversion rate**: A conversion refers to the moment when a user responds to a call to action. This rate is the percentage of users who have entered the funnel and converted.

  **Note**: Only completed RUM sessions contribute to conversion rates.

> - **Time to convert**: The time it took for the user to complete the step events.

Reviewer 2: Note in case you want to add it somewhere: This is not available for the "steps" viz, it will switch to timeseries when selected

{{< img src="product_analytics/journeys/funnel_analysis/pana_funnel_toplist.png" alt="View conversion data as a top list." style="width:80%;" >}}

> ## Navigate funnel charts

Reviewer: It doesn't have to be now, but I would think a bit about how you want to order the information on this page. Right now, it looks like we have some broad categories of:
- Interacting with visualizations (building a funnel, changing the visualization type, filtering, changing conversion calculation methods)
- Conceptual stuff (defining funnel analysis and conversions)
- Gotchas (telling users when unexpected behaviour might happen)

It is a lot to figure out which order will progressively build up users' knowledge without confusing them! I recommend doing as little jumping around as possible to help users feel like they're going from one chapter in a book to another. This section made me think of it because I thought it might be more appropriate closer to the section where you're building the chart, so you could tell customers to build the chart, and then tell them how to navigate it.
